app/models/attachment_reference.rb

```diff
@@ -0,0 +1,57 @@
+class AttachmentReference < ActiveRecord::Base
+  after_initialize :set_expiry_at, if: :new_record?
```

Comment:
there's a slight problem with this, if the initialize is called with an owner, this will have both owner and expiry.

solution is probably to have the after_initialize check for owner.